DigitalLearningSolutions.Web/Controllers/MyAccountController.cs

```diff
@@ -239,8 +249,29 @@ DlsSubApplication dlsSubApplication
                 delegateAccount?.Id
             );
 
+            var modifiedEmails = new List<(string, int?)>();

```

Comment:
We should make changes to this method, because it is now _massive_
We should split it into smaller methods, and move as many of them as we can into a new MyAccount service
* Validate data requirements and formatting
* If invalid return
* Validate email uniqueness
* If invalid return
* Get list of changed emails
* Save changes
* Send the user on

Reply:
I'd also like to separate the centreless scenario from the logged-into-centre scenario a bit more strongly, rather than repeatedly checking which scenario you're in. Probably branch after doing validation.

Reply:
I split that huge method into several smaller ones, but I didn't move them in a service, I don't see the point of doing that. The controller will still call all those methods and will have the same level of complexity. We will not call those methods in another controller, so I don't see the difference between having those methods in a service and having them in the controller.